vsphere/internal/helper/role/role_helper.go

```diff
@@ -0,0 +1,401 @@
+	"VirtualMachine.GuestOperations.ModifyAliases",
+	"VirtualMachine.GuestOperations.Query",
+	"VirtualMachine.GuestOperations.QueryAliases",
+	"VirtualMachine.Hbr.ConfigureReplication",
+	"VirtualMachine.Hbr.MonitorReplication",
+	"VirtualMachine.Hbr.ReplicaManagement",
+	"VirtualMachine.Interact.AnswerQuestion",
+	"VirtualMachine.Interact.Backup",
+	"VirtualMachine.Interact.ConsoleInteract",
+	"VirtualMachine.Interact.CreateScreenshot",
+	"VirtualMachine.Interact.CreateSecondary",
+	"VirtualMachine.Interact.DefragmentAllDisks",
+	"VirtualMachine.Interact.DeviceConnection",
+	"VirtualMachine.Interact.DisableSecondary",
+	"VirtualMachine.Interact.DnD",
+	"VirtualMachine.Interact.EnableSecondary",
+	"VirtualMachine.Interact.GuestControl",
+	"VirtualMachine.Interact.MakePrimary",
+	"VirtualMachine.Interact.Pause",
+	"VirtualMachine.Interact.PowerOff",
+	"VirtualMachine.Interact.PowerOn",
+	"VirtualMachine.Interact.PutUsbScanCodes",
+	"VirtualMachine.Interact.Record",
+	"VirtualMachine.Interact.Replay",
+	"VirtualMachine.Interact.Reset",
+	"VirtualMachine.Interact.SESparseMaintenance",
+	"VirtualMachine.Interact.SetCDMedia",
+	"VirtualMachine.Interact.SetFloppyMedia",
+	"VirtualMachine.Interact.Suspend",
+	"VirtualMachine.Interact.TerminateFaultTolerantVM",
+	"VirtualMachine.Interact.ToolsInstall",
+	"VirtualMachine.Interact.TurnOffFaultTolerance",
+	"VirtualMachine.Inventory.Create",
+	"VirtualMachine.Inventory.CreateFromExisting",
+	"VirtualMachine.Inventory.Delete",
+	"VirtualMachine.Inventory.Move",
+	"VirtualMachine.Inventory.Register",
+	"VirtualMachine.Inventory.Unregister",
+	"VirtualMachine.Namespace.Event",
+	"VirtualMachine.Namespace.EventNotify",
+	"VirtualMachine.Namespace.Management",
+	"VirtualMachine.Namespace.ModifyContent",
+	"VirtualMachine.Namespace.Query",
+	"VirtualMachine.Namespace.ReadContent",
+	"VirtualMachine.Provisioning.Clone",
+	"VirtualMachine.Provisioning.CloneTemplate",
+	"VirtualMachine.Provisioning.CreateTemplateFromVM",
+	"VirtualMachine.Provisioning.Customize",
+	"VirtualMachine.Provisioning.DeployTemplate",
+	"VirtualMachine.Provisioning.DiskRandomAccess",
+	"VirtualMachine.Provisioning.DiskRandomRead",
+	"VirtualMachine.Provisioning.FileRandomAccess",
+	"VirtualMachine.Provisioning.GetVmFiles",
+	"VirtualMachine.Provisioning.MarkAsTemplate",
+	"VirtualMachine.Provisioning.MarkAsVM",
+	"VirtualMachine.Provisioning.ModifyCustSpecs",
+	"VirtualMachine.Provisioning.PromoteDisks",
+	"VirtualMachine.Provisioning.PutVmFiles",
+	"VirtualMachine.Provisioning.ReadCustSpecs",
+	"VirtualMachine.State.CreateSnapshot",
+	"VirtualMachine.State.RemoveSnapshot",
+	"VirtualMachine.State.RenameSnapshot",
+	"VirtualMachine.State.RevertToSnapshot",
+	"vService.CreateDependency",
+	"vService.DestroyDependency",
+	"vService.ReconfigureDependency",
+	"vService.UpdateDependency",
+}
+
+// ExistsByID check if a role exist using id, and return that role
+func ExistsByID(client *govmomi.Client, id int) (*types.AuthorizationRole, error) {
```

Comment:
To match the style of other resources, and because these do more than just check if the role exists, I think `ByID` and `ByName` would be better.